tests/failpoints/cases/test_split_region.rs

```diff
@@ -552,6 +552,90 @@ fn test_split_not_to_split_existing_tombstone_region() {
     must_get_equal(&cluster.get_engine(2), b"k1", b"v1");
 }
 
+#[test]
+fn test_snap_handling_after_peer_is_replaced_by_split_and_removed() {
+    // The following diagram represents the final state of the test:
+    //
+    //                    ┌───────────┐  ┌───────────┐  ┌───────────┐
+    //                    │           │  │           │  │           │
+    //      Region 1      │ Peer 1    │  │ Peer 2    │  │ Peer 3    │
+    //      [k2, +∞)      │           │  │           │  │           │
+    // ───────────────────┼───────────┼──┼───────────┼──┼───────────┼──
+    //                    │           │  │           │  │           │
+    //      Region 1000   │ Peer 1001 │  │ Peer 1003 │  │ Peer 1002 │
```

Comment:
```suggestion
    //      Region 1000   │ Peer 1001 │  │ Peer 1002 │  │ Peer 1003 │
```

I think 1003 is in store3？

Reply:
Store 2 actually had peer 1003 when I examined the logs. I don't know why it turned out that way in tests.